### PR TITLE
fix: replace fuse.js with leven to fix no similar matching name error

### PR DESCRIPTION
**Pull request by cehan-Chloe**
fix for https://github.com/player-ui/player/issues/450 

### Change Type (required)
Indicate the type of change your pull request is:

- [ ] `patch`
- [ ] `minor`
- [ ] `major`

### Does your PR have any documentation updates?
- [ ] Updated docs
- [x] No Update needed
- [ ] Unable to update docs

<details>
  <summary>📦 Published PR as canary version: <code>0.7.5--canary.452.15858</code></summary>
  <br />

  Try this version out locally by upgrading relevant packages to 0.7.5--canary.452.15858
</details>

### CODE REVIEW

**Comment by cehan-Chloe** [comment]
| https://bundlephobia.com/package/@player-ui/react@0.7.5--canary.452.15858    | https://bundlephobia.com/package/@player-ui/react@0.8.0-next.4 |
| -------- | ------- |
| image  | image    |

File: react/player/src/asset/index.tsx [hunk 1]
```diff
@@ -66,11 +66,16 @@ export const ReactAsset = (
...
 
-    const fuse = new Fuse(matchList, { keys: ["type", "key"] });
-    const similarType = JSON.stringify(
-      fuse.search(unwrapped.type as string)[0].item,
+    const typeList = matchList.map(
+      (match) => JSON.parse(JSON.stringify(match)).type,
     );
 
+    const similarType = typeList.reduce((prev, curr) => {
+      return leven(unwrapped.type, prev) < leven(unwrapped.type, curr)
```
Comment (adierkens) [inline react/player/src/asset/index.tsx:74]: _nit_: can we store the value of the previous `leven` call, to duplication?

Comment (cehan-Chloe) [reply react/player/src/asset/index.tsx:74]:
````
    let minDistance = leven(unwrapped.type as string, typeList[0]);
    let similarType = typeList[0];
    typeList.forEach(type => {
      const distance = leven(unwrapped.type, type);
      if (distance < minDistance) {
        minDistance = distance;
        similarType = type;
      }
    });
````
when you say storing prev value, do you mean this?

Comment (adierkens) [reply react/player/src/asset/index.tsx:74]: You can still keep everything in a single `reduce()`, but the way it's structured now we end up calling `leven()` for the previous values that we have already computed. 

something like: 
```ts
typeList.reduce((prev, curr) => {
  const next = {
    value: leven(unwrapped.type, curr),
    type: unwrapped.type
  };

  if (prev === undefined || prev.value < next.value) {
    return prev;
  }

  return next;
}, undefined);
```

**Review by adierkens — APPROVED** [review]

**Pull request merged**